TM1py/Services/ElementService.py

```diff
@@ -659,14 +664,22 @@ def _get_mdx_set_cardinality(self, mdx: str) -> int:
         response = self._rest.POST(url, json.dumps(payload, ensure_ascii=False))
         return response.json()['Cardinality']
 
-    def _build_drill_intersection_mdx(self, dimension_name: str, hierarchy_name: str, first_element_name: str,
-                          second_element_name: str, recursive: bool) ->str:
+    @staticmethod
+    def _build_drill_intersection_mdx(dimension_name: str, hierarchy_name: str, first_element_name: str,
+                                      second_element_name: str, mdx_method: str, recursive: bool) -> str:
 
         first_member = Member.of(dimension_name, hierarchy_name, first_element_name)
         second_member = Member.of(dimension_name, hierarchy_name, second_element_name)
-        mdx = MdxHierarchySet.members([first_member]).tm1_drill_down_member(all=True, recursive=recursive)\
-            .intersect(MdxHierarchySet.members([second_member])).to_mdx()
 
+        if not hasattr(MDXDrillMethod, mdx_method.upper()):
+            raise TM1pyException('Invalid MDX Drill Method Specified, Options: TM1DrillDownMember or Descendants')
+        elif MDXDrillMethod.DESCENDANTS.name == mdx_method.upper():
+            mdx = MdxHierarchySet.members([first_member]).tm1_drill_down_member(all=True, recursive=recursive) \
```

Comment:
@rclapp I think we should use `descendants` here in this condition and `tm1_drill_down_member` in the elif below.